processing/src/main/java/io/druid/segment/DoubleColumnSelector.java

```diff
@@ -26,5 +26,32 @@
 public interface DoubleColumnSelector extends ColumnValueSelector, HotLoopCallee
 {
   @CalledFromHotLoop
-  double get();
+  @Override
+  double getDouble();
```

Comment:
> Also "As" method names are longer, and I hardly think anyone could be confused by the current names (e. g. think about them as they have ByteBuffer.getDouble() semantics). So I'm slightly pro keeping the current names.

I was confused because I thought ```DoubleColumnSelector``` is specialized for the double type and will have a method to retrieve double values like ```java.nio.DoubleBuffer``` has a series of ```get()``` which return only doubles.

I looked into more codes around index merger and understand why you chose this design. I don't see any better design for now. Please go for it.